src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,259 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from multiprocessing import Queue
+import threading
+
+from avocado.core.exceptions import TestFail
+from dmg_utils import get_storage_query_device_info, get_dmg_response
+from exception_utils import CommandFailure
+from ior_utils import run_ior, thread_run_ior
+from job_manager_utils import get_job_manager
+from led import VmdLedStatus
+from nvme_utils import set_device_faulty
+
+
+class NvmeFaultReintegrate(VmdLedStatus):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description: This test runs
+    NVME fault and reintegration test cases.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up for test case."""
+        super().setUp()
+        self.daos_command = self.get_daos_command()
+
+    def verify_dev_led_state(self, device, dev_state="NORMAL", led_state="OFF"):
+        """Verify device dev_state and led_state.
+
+        Args:
+            device (str): the device description.
+            dev_state (str): expect dev_state, default to "NORMAL".
+            led_state (str): expect led_state, default to "OFF".
+
+        Returns:
+            bool: True if the expected dev_state and led_state.
+
+        """
+        return self.check_result(self.get_led_status_value(device), dev_state, led_state)
+
+    def check_result(self, result, dev_state, led_state):
+        """Check for result of storage device and led states.
+
+        Args:
+            result (dict): return from get_led_status_value for a storage device.
+            dev_state (str): expect dev_state.
+            led_state (str): expect led_state.
+
+        Return:
+            True if the expected dev_state and led_state.
+
+        """
+        if "response" in result.keys():
+            for value in list(result['response']['host_storage_map'].values()):
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            return True
+        elif "host_storage_map" in result.keys():
+            for value in list(result['host_storage_map'].values()):
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            return True
+        else:
+            self.log.info("##Unsupported result dictionary %s", result)
```

Comment:
All code calling this function ends up calling `self.fail()` when this method returns `False`, so maybe not?  Either way it would be good to log some more information when this happens:
```suggestion
            self.log.debug("Neither 'response' nor 'host_storage_map' found in dmg command json output:")
            self.log.debug('  %s', result)
```
Then we could optionally fail at this point or let the caller fail the test:
```
            self.fail('Error checking dmg command json output')
```
@daltonbohning thoughts?